code/game/objects/items/weapons/RCD.dm

```diff
@@ -151,9 +155,19 @@
 	else
 		qdel(T)
 
+	update_icon()	//Updates the ammo counter if ammo is succesfully used
```

Comment:
Why not move `update_icon()` in `useResource(...)` proc?

Reply:
I'll try that and see if it works.

Reply:
Thank. Will wait for result.